### PR TITLE
[Desing SIG] Modularize user_accounts

### PR DESCRIPTION
**Pull request by xlecours**
This pull request make the instrument manager module comply to the new Module framework. It adds namespacing and rely on the Module class to load.

### CODE REVIEW

File: modules/user_accounts/php/edit_user.class.inc [hunk 1]
```diff
@@ -416,58 +417,6 @@ class NDB_Form_User_Accounts extends NDB_Form
...
-                                        )
-                                    );
-                                }
-                            }
-                            unset($values[$var_name]);
-                        }
-                    }
-                }
-            }
-        }
```
Comment (driusan) [inline modules/user_accounts/php/edit_user.class.inc]: can't comment on it, but I think the line below here is meant to be removed (you removed the "START NOTIFICATIONS" comment at the start..)

File: modules/user_accounts/php/edit_user.class.inc [hunk 2]
```diff
@@ -1328,7 +1106,7 @@ class NDB_Form_User_Accounts extends NDB_Form
         ) {
             // If PHP version < 5.5, use old MD5Unsalt method to check for
             // password change
-            return !User::MD5Unsalt($newPassword, $passwords['Password_hash']);
+            return !\User::MD5Unsalt($newPassword, $passwords['Password_hash']);
```
Comment (driusan) [inline modules/user_accounts/php/edit_user.class.inc]: can you make a redmine ticket to delete this? I thought it was all gone.. the password_md5 column doesn't even exist anymore, so I think this line is just a relic of the past.

File: modules/user_accounts/php/my_preferences.class.inc [hunk 3]
```diff
@@ -0,0 +1,1134 @@
...
+
+        ////Get the current permissions/////
+        $current_permissionids = $user->getPermissionIDs();
+
+        $permIDs = array();
+        // store the permission IDs
+        if (!empty($values['permID'])) {
+            $permIDs = array_keys($values['permID']);
+        }
+        unset($values['permID']);
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: all the stuff related to permissions should be taken out of the my_preferences page too.

File: modules/user_accounts/php/my_preferences.class.inc [hunk 4]
```diff
@@ -0,0 +1,1134 @@
...
+        unset($values['NA_Password']);
+
+        // If editing a user and nothing was specified in the password text field
+        // remove Password_md5 from the value set, otherwise Password_md5
+        // will be set to '' by the system
+        if ($values['Password_hash'] == '' && !$this->isCreatingNewUser()) {
+            unset($values['Password_hash']);
+        }
+
+        if ($this->page === 'edit_user') {
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: this is always false

File: modules/user_accounts/php/my_preferences.class.inc [hunk 5]
```diff
@@ -0,0 +1,1134 @@
...
+                    );
+                }
+            }
+            unset($values['examiner_radiologist']);
+            unset($values['examiner_pending']);
+            //END EXAMINER UPDATE
+        }
+
+        // START NOTIFICATIONS UPDATE
+        if ($this->page==='my_preferences') {
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: this is always true.

File: modules/user_accounts/php/my_preferences.class.inc [hunk 6]
```diff
@@ -0,0 +1,1134 @@
...
+                ? $values['Password_expiry'] : null;
+            $user->updatePassword($set['Password_hash'], $expiry);
+        }
+
+        // update the user permissions if applicable
+        // If the user is editing his/her own account, skip the part where
+        // changes to the permissions are handled (there should not be any
+        // change to the user's permission set since all the checkboxes are
+        // disabled)
+        if (!$this->_isEditingOwnAccount()) {
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: the permissions part should never be applicable, I think this whole block can be removed.

File: modules/user_accounts/php/my_preferences.class.inc [hunk 7]
```diff
@@ -0,0 +1,1134 @@
...
+            $msg_data['password'] = $values['Password_hash'];
+
+            $template = (is_null($this->identifier))
+                ? 'new_user.tpl' : 'edit_user.tpl';
+            \Email::send($values['Email'], $template, $msg_data);
+        }
+
+        $this->tpl_data['success'] = true;
+
+        if ($this->isCreatingNewUser()) {
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: always false

File: modules/user_accounts/php/my_preferences.class.inc [hunk 8]
```diff
@@ -0,0 +1,1134 @@
...
+                }
+            }
+
+            if (!empty($permIDs)) {
+                $user->addPermissions($permIDs);
+            }
+        }
+
+        // send the user an email
+        if (!empty($send)) {
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: is this necessary for changing preferences?

File: modules/user_accounts/php/my_preferences.class.inc [hunk 9]
```diff
@@ -0,0 +1,1134 @@
...
+     * exist and account is still pending approval, to ensure the
+     * users with account activity can't be rejected by setting
+     * their status to pending.
+     *
+     * @param string $id identifier of account to verify
+     *
+     * @return boolean true is Password_hash is null and
+     * Pending_approval is Yes, false otherwsie
+     */
+    function _canRejectAccount($id)
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: shouldn't be applicable to preferences

File: modules/user_accounts/php/my_preferences.class.inc [hunk 10]
```diff
@@ -0,0 +1,1134 @@
...
+        return false;
+    }
+
+    /**
+     * Controls the output/behaviour of the form is used to edit
+     * the user's preferences.
+     *
+     * @return void
+     */
+    // @codingStandardsIgnoreStart
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: why is this ignoring coding standards?

File: modules/user_accounts/php/my_preferences.class.inc [hunk 11]
```diff
@@ -0,0 +1,1134 @@
...
+
+
+    /**
+     * Validates the data entered in the edit user form.
+     *
+     * @param array $values what the user entered on the form.
+     *
+     * @return array $errors all the errors found.
+     */
+    function _validateEditUser($values)
```
Comment (driusan) [inline modules/user_accounts/php/my_preferences.class.inc]: this is for the other page too.

File: modules/user_accounts/php/edit_user.class.inc (hunk 2 above)
Comment (xlecours) [reply modules/user_accounts/php/edit_user.class.inc]: Done RM#13404

**Comment by ridz1208** [comment]
@xlecours @driusan if this is going to 19.0 it will have to be rebased on https://github.com/aces/Loris/pull/2642

File: modules/user_accounts/php/my_preferences.class.inc (hunk 6 above)
Comment (ridz1208) [reply modules/user_accounts/php/my_preferences.class.inc]: agree

File: modules/user_accounts/php/my_preferences.class.inc (hunk 8 above)
Comment (ridz1208) [reply modules/user_accounts/php/my_preferences.class.inc]: no

File: modules/user_accounts/php/my_preferences.class.inc (hunk 7 above)
Comment (ridz1208) [reply modules/user_accounts/php/my_preferences.class.inc]: agree

File: modules/user_accounts/php/my_preferences.class.inc (hunk 9 above)
Comment (ridz1208) [reply modules/user_accounts/php/my_preferences.class.inc]: agree

File: modules/user_accounts/php/my_preferences.class.inc (hunk 10 above)
Comment (ridz1208) [reply modules/user_accounts/php/my_preferences.class.inc]: becauseloris

**Comment by ridz1208** [comment]
@xlecours  I will test/review once first batch of changes changes is done

**Comment by xlecours** [comment]
There is a better way of doing this.

**Pull request closed without merging**